packages/api-explorer/e2e/e2e.spec.ts

```diff
@@ -0,0 +1,237 @@
+/*
+
+ MIT License
+
+ Copyright (c) 2021 Looker Data Sciences, Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+import '@testing-library/jest-dom'
+
+import { getSpecKey, goToPage, pageReload } from './helpers'
+
+// https://github.com/smooth-code/jest-puppeteer/tree/master/packages/expect-puppeteer
+// https://github.com/puppeteer/puppeteer/blob/main/docs/api.md
+
+jest.setTimeout(120000)
+
+const BASE_URL = 'https://localhost:8080'
+
+describe('API Explorer', () => {
+  beforeEach(async () => {
+    await jestPuppeteer.resetBrowser()
+  })
+
+  describe('general', () => {
+    beforeEach(async () => {
+      await page.goto(BASE_URL)
+    })
+    it('renders a method page', async () => {
+      await expect(page).toClick('h5', { text: 'Dashboard' })
+      await expect(page).toClick('a', { text: 'Get All Dashboards' })
+      const specKey = await getSpecKey()
+      await expect(page.url()).toEqual(
+        `${BASE_URL}/${specKey}/methods/Dashboard/all_dashboards`
+      )
```

Comment:
With everything being async here is there a significant performance difference between puppeteer and cypress?

Reply:
if anything i'd be concerned about cypress using blocking I/O